src/draw/border.c

```diff
@@ -1,886 +1,886 @@
-	bound.x = box->x;
-	bound.width = bl_width;
-	bound.height = bl_height;
-	bound.y = box->y + box->height - bound.height;
-	if (LCUIRect_GetOverlayRect(&bound, &paint->rect, &rect)) {
-		bound_left = bound.x - rect.x;
-		bound_top = bound.y - rect.y;
-		rect.x -= paint->rect.x;
-		rect.y -= paint->rect.y;
-		Graph_Quote(&canvas, &paint->canvas, &rect);
-		DrawBorderBottomLeft(&canvas, bound_left, bound_top,
-				     &border->bottom, &border->left,
-				     border->bottom_left_radius);
-	}
-	/* Draw border bottom right angle */
-	bound.width = br_width;
-	bound.height = br_height;
-	bound.x = box->x + box->width - bound.width;
-	bound.y = box->y + box->height - bound.height;
-	if (LCUIRect_GetOverlayRect(&bound, &paint->rect, &rect)) {
-		bound_left = bound.x - rect.x;
-		bound_top = bound.y - rect.y;
-		rect.x -= paint->rect.x;
-		rect.y -= paint->rect.y;
-		Graph_Quote(&canvas, &paint->canvas, &rect);
-		DrawBorderBottomRight(&canvas, bound_left, bound_top,
-				      &border->bottom, &border->right,
-				      border->bottom_right_radius);
-	}
-	/* Draw top border line */
-	bound.x = box->x + tl_width;
-	bound.y = box->y;
-	bound.width = box->width - tl_width - tr_width;
-	bound.height = border->top.width;
-	if (LCUIRect_GetOverlayRect(&bound, &paint->rect, &bound)) {
-		bound.x -= paint->rect.x;
-		bound.y -= paint->rect.y;
-		Graph_Quote(&canvas, &paint->canvas, &bound);
-		Graph_FillRect(&canvas, border->top.color, NULL, TRUE);
-	}
-	/* Draw bottom border line */
-	bound.x = box->x + bl_width;
-	bound.y = box->y + box->height - border->bottom.width;
-	bound.width = box->width - bl_width - br_width;
-	bound.height = border->bottom.width;
-	if (LCUIRect_GetOverlayRect(&bound, &paint->rect, &bound)) {
-		bound.x -= paint->rect.x;
-		bound.y -= paint->rect.y;
-		Graph_Quote(&canvas, &paint->canvas, &bound);
-		Graph_FillRect(&canvas, border->bottom.color, NULL, TRUE);
-	}
-	/* Draw left border line */
-	bound.y = box->y + tl_height;
-	bound.x = box->x;
-	bound.width = border->left.width;
-	bound.height = box->height - tl_height - bl_height;
-	if (LCUIRect_GetOverlayRect(&bound, &paint->rect, &bound)) {
-		bound.x -= paint->rect.x;
-		bound.y -= paint->rect.y;
-		Graph_Quote(&canvas, &paint->canvas, &bound);
-		Graph_FillRect(&canvas, border->left.color, NULL, TRUE);
-	}
-	/* Draw right border line */
-	bound.x = box->x + box->width - border->right.width;
-	bound.y = box->y + tr_height;
-	bound.width = border->right.width;
-	bound.height = box->height - tr_height - br_height;
-	if (LCUIRect_GetOverlayRect(&bound, &paint->rect, &bound)) {
-		bound.x -= paint->rect.x;
-		bound.y -= paint->rect.y;
-		Graph_Quote(&canvas, &paint->canvas, &bound);
-		Graph_FillRect(&canvas, border->right.color, NULL, TRUE);
-	}
-	return 0;
-}
+﻿/*
+ * border.c -- Border drawing
+ *
+ * Copyright (c) 2018-2019, Liu chao <lc-soft@live.cn> All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
```

Comment:
There are too many changes to this file, have you changed the linebreak style (`\r\n` -> `\n`) of this file?

Reply:
thx, fixed that up